include/tunnler/room_member.h

```diff
@@ -160,6 +166,7 @@ class RoomMember final {
     // cause an overflow in UDS::RecvBeaconBroadcastData.
     static const size_t MaxBeaconQueueSize = 25;
 
+    std::mutex chat_mutex;             ///< Mutex to protect access to the chat queue.
     std::deque<ChatEntry> chat_queue;    ///< List of all chat messages received since last PopChatEntries was called
```

Comment:
If we do that we should also move the wifi packet handling to the users. I don't really have an opinion about this.